src/thumbnailer.h

```diff
@@ -84,14 +87,43 @@ class Thumbnailer {
 
   Status SetLoopCount(WebPData* const webp_data);
 
+  // Generates the animation with the slope optimization for the RD-curve.
+  // Either lossy and near-lossless compression modes will be used for each
+  // frame.
+  Status GenerateAnimationSlopeOptim(WebPData* const webp_data);
```

Comment:
Can you add some tests for this method, at least to make sure valgrind catches any memory leaks.

Reply:
Yes, I will.